### PR TITLE
(don't merge) Upgrade for 2.0

### PR DESCRIPTION
**Pull request by beckysiegel**
Port to 2.0

### CODE REVIEW

**Comment by beckysiegel** [comment]
@notwaldorf there are two sort of strange things happening on the mainline demo
 1) the iron-icons in the demo 404 because the path seems to be calculated incorrectly. (It is looking for /components/gold-cc-cvc-input/demo/cvc_hint.png but the image is not inside the demo directory), as opposed to /components/gold-cc-cvc-input/cvc_hint.png in v1.x

2) In the demo, it doesn't seem like the iron-flex property is applied to the css rule somehow, but no idea why it's not working.

screen shot 2017-05-05 at 2 06 49 pm

screen shot 2017-05-05 at 2 06 57 pm

File: gold-cc-cvc-input.html [hunk 1]
```diff
@@ -173,9 +239,44 @@
...
+        beforeRegister: function() {
+          this._beforeRegister();
+        },
+
+        _beforeRegister: function() {
+          var template = Polymer.DomModule.import('gold-cc-cvc-input', 'template');
+          var version = Polymer.Element ? 'v1' : 'v0';
+          var inputTemplate = Polymer.DomModule.import('gold-cc-cvc-input', 'template#' + version);
+          var inputPlaceholder = template.content.querySelector('#template-placeholder');
+          inputPlaceholder.parentNode.replaceChild(inputTemplate.content, inputPlaceholder);
```
Comment (e111077) [inline gold-cc-cvc-input.html]: I think to keep this element extensible this should be wrapped in an `if (inputPlaceholder)` like in [`paper-input`](https://github.com/PolymerElements/paper-input/blob/master/paper-input.html#L267)

File: gold-cc-cvc-input.html [hunk 2]
```diff
@@ -224,6 +325,9 @@
          * Overidden from Polymer.IronControlState.
          */
         _onFocusedChanged: function(focused) {
+          if (!this._focusableElement) {
```
Comment (e111077) [inline gold-cc-cvc-input.html]: When is this conditional triggered?

File: bower.json [hunk 3]
```diff
@@ -21,18 +21,37 @@
   "homepage": "https://github.com/PolymerElements/gold-cc-cvc-input",
   "ignore": [],
   "dependencies": {
-    "iron-icon": "PolymerElements/iron-icon#^1.0.0",
-    "iron-flex-layout": "PolymerElements/iron-flex-layout#^1.0.0",
-    "iron-form-element-behavior": "PolymerElements/iron-form-element-behavior#^1.0.0",
-    "paper-input": "PolymerElements/paper-input#^1.0.0",
-    "paper-styles": "PolymerElements/paper-styles#^1.0.0",
-    "polymer": "Polymer/polymer#^1.1.0"
+    "iron-icon": "PolymerElements/iron-icon#2.0-preview",
```
Comment (e111077) [inline bower.json]: can you please update these to `1 - 2`, `1.9 - 2` and add the resolutions etc.

[paper-input](https://github.com/PolymerElements/paper-input/blob/master/bower.json) has a good example

**Comment by e111077** [comment]
Performed a rebase. Waiting on tests...

**Pull request merged**